#!/usr/bin/env python
PACKAGE = "potbot_pcl"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("DownSampling_voxel_size",    double_t, 0, "A double parameter",      0.03, 0.01, 1)

gen.add("PlaneRemoval_distance_threshold",    double_t, 0, "A double parameter",      0.02, 0.01, 1)

gen.add("Euclidean_cluster_tolerance",          double_t, 0, "A double parameter",      0.5, 0, 1)
gen.add("Euclidean_min_cluster_size",           int_t,    0, "An Integer parameter",    100, 1, 200)

gen.add("Supervoxel_voxel_resolution",          double_t, 0, "A double parameter",      0.1, 0, 1)
gen.add("Supervoxel_seed_resolution",           double_t, 0, "A double parameter",      0.3, 0.01, 1)
gen.add("Supervoxel_color_importance",          double_t, 0, "A double parameter",      0.2, 0, 1)
gen.add("Supervoxel_spatial_importance",        double_t, 0, "A double parameter",      0.4, 0, 1)
gen.add("Supervoxel_normal_importance",         double_t, 0, "A double parameter",      1.0, 0, 1)

exit(gen.generate(PACKAGE, "potbot", "ClusteringParam"))